lib/Parse/ParseDecl.cpp

```diff
@@ -4148,6 +4206,13 @@ Parser::parseDecl(ParseDeclOptions Flags,
   // Obvious nonsense.
   default:
 
+    if (shouldParseExperimentalConcurrency() && Tok.is(tok::identifier) &&
```

Comment:
You don't need the `Tok.is(too::identifier)` here, because it's part of `Tok.isContextualKeyword`.